src/common/test-json.c

```diff
@@ -238,6 +239,39 @@ test_get_array (TestCase *tc,
 }
 
 static void
+test_get_object (TestCase *tc,
+                gconstpointer data)
+{
+  JsonObject *defawlt = json_object_new ();
+  gboolean ret;
+  JsonObject *value;
+
+  ret = cockpit_json_get_object (tc->root, "object", NULL, &value);
+  g_assert (ret == TRUE);
+  g_assert (value != NULL);
+  g_assert_cmpstr (json_object_get_string_member (value, "test"), ==, "one");
+
+  ret = cockpit_json_get_object (tc->root, "object", NULL, NULL);
+  g_assert (ret == TRUE);
+
+  ret = cockpit_json_get_object (tc->root, "unknown", NULL, &value);
+  g_assert (ret == TRUE);
+  g_assert (value == NULL);
+
+  ret = cockpit_json_get_object (tc->root, "unknown", defawlt, &value);
+  g_assert (ret == TRUE);
+  g_assert (value == defawlt);
+
+  ret = cockpit_json_get_object (tc->root, "number", NULL, &value);
+  g_assert (ret == FALSE);
+
+  ret = cockpit_json_get_object (tc->root, "array", NULL, NULL);
+  g_assert (ret == FALSE);
+
```

Comment:
For the sake of completeness we should add a test here for the case "query existing entry, pass default value, result is not the default value."
This shouldn't be an issue, but would improve the coverage.